### DISCUSSION TITLE
Comparação de números em Arrays (Java Script)

### DISCUSSION THREAD

**Discussion by rdiego09** [frontendbr/forum#1895] [Perguntas e Respostas]
Olá pessoal, tudo bem? 

Estou começando meus estudos em programação e estou tentando criar um código de simulação de loteria. 
Porém não estou conseguindo realizar a comparação entre os números escolhidos e os números sorteados. 
Por exemplo, gostaria que aparecesse uma mensagem com os números que a pessoa acertou. 

Segue o pedaço do código que estou tentando fazer isso acontecer:
___________________________________________________________________________________________
<script>
var numerosDigitados = [1 ,2 ,3, 4, 5, 6]
var numerosSorteados = [6, 7, 8, 9, 10, 11]

for(var j=0; j < numerosDigitados.length; j++) {
		for(var k=0; k < numerosSorteados.length; k++) {
			if(digitado[j] == numerosSorteados[k]) {
				alert("Você acertou o número " numerosDigitado[j]);
			}
			else {
			}	alert("Você não acertou nenhum número");
		}
	}

</script>
_________________________________________________________________________________________

Desde já agradeço quem puder ajudar.

Abraço!

**Comment by victorferreira** [comment]
Dá uma olhada nos métodos [`filter`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) e [`includes`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) do objeto [`Array`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array)

```
var numerosDigitados = [1, 2, 3, 4, 5, 6];
var numerosSorteados = [6, 7, 8, 9, 10, 11];
var numerosAcertados = numerosDigitados.filter(num => numerosSorteados.includes(num));
```

**Comment by rdiego09** [comment]
Muito obrigado Victor!

Não conhecia esses métodos. Muito mais fácil assim heheheh

Só não entendi pq foi usado o símbolo =>. Pq não da certo se eu usar o ==?

Valeu pela ajuda

**Comment by victorferreira** [comment]
O método _filter_ aceita uma função como parâmetro que deve ser executada para cada item do array. Essa sintaxe é apenas uma outra forma de escrever uma função. Alternativamente, você poderia usar o código abaixo e ter o mesmo resultado:

```
var numerosAcertados = numerosDigitados.filter(function(num) {
    return numerosSorteados.includes(num);
});
```

[Documentação sobre _arrow functions_](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Functions/Arrow_functions).

**Comment by rdiego09** [comment]
Consegui entender agora brother.

Muito obrigado pela força!
